core/camel-base-engine/src/main/java/org/apache/camel/impl/engine/AbstractCamelContext.java

```diff
@@ -4547,10 +4550,28 @@ public XMLRoutesDefinitionLoader getXMLRoutesDefinitionLoader() {
         return xmlRoutesDefinitionLoader;
     }
 
+    @Override
     public void setXMLRoutesDefinitionLoader(XMLRoutesDefinitionLoader xmlRoutesDefinitionLoader) {
         this.xmlRoutesDefinitionLoader = doAddService(xmlRoutesDefinitionLoader);
     }
 
+    @Override
+    public RoutesLoader getRoutesLoader() {
```

Comment:
We should warm up the routes loader in doStartEagerServices like we do for the other services we really want to start early